File: src/2810-Faulty-Keyboard/faulty_keyboard.cpp
#include <iostream>
#include <string>
#include <algorithm>

class Solution {
public:
    std::string finalString(const std::string& s) {
        std::string ans = "";

        for (char c : s) {
            if (c == 'i') {
                std::reverse(ans.begin(), ans.end());
            } else {
                ans += c;
            }
        }

        return ans;
    }
};

int main() {
    std::string s = "string";
    Solution solution;

    std::cout << solution.finalString(s) << std::endl;

    return 0;
}

File: src/2810-Faulty-Keyboard/faulty_keyboard.py
class Solution:
    def finalString(self, s: str) -> str:
        ans: str = ""

        for c in s:
            if c == 'i':
                ans = ans[::-1]
            else:
                ans += c
        
        return ans
    
if __name__ == "__main__":
    s: str =  "string"
    solution = Solution()
    
    print(solution.finalString(s))